### PR TITLE
Added PhP easter egg module

### PR DESCRIPTION
**Pull request by aman566**
#### Checklist
- [X] I have followed the [Contributor Guidelines](https://github.com/zdresearch/OWASP-Nettacker/wiki/Developers#contribution-guidelines).
- [X] I have added the relevant documentation.
- [X] My branch is up-to-date with the Upstream master branch.

#### Changes proposed in this pull request
- [X] Added the php easter egg module in order to detect some phpinfo() pages or the php version from the gif image displayed
- [X] Uses the useragents from the lib/payload/wordlists library
#### Your development environment
- OS: `Linux`
- OS Version: `Ubuntu 18.04`
- Python Version: `3.6.9`

### CODE REVIEW

File: lib/vuln/php_easter_egg/engine.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+            data = json.dumps({'HOST': target, 'USERNAME': '', 'PASSWORD': '', 'PORT': '', 'TYPE': 'php_easter_eggs_vuln_scan',
+                               'DESCRIPTION': messages(language, "not_found"), 'TIME': now(),
+                               'CATEGORY': "scan", 'SCAN_ID': scan_id, 'SCAN_CMD': scan_cmd})
+            __log_into_file(log_in_file, 'a', data, language)
+        os.remove(thread_tmp_filename)
+
+    else:
+        print("aman")
+        warn(messages(language, "input_target_error").format(
+            'php_easter_eggs_vuln_scan', target))
```
Comment (securestep9) [inline lib/vuln/php_easter_egg/engine.py:201]: @aman566  should be '_egg' not '_eggs' - please stay consistent and pay attention to details

File: lib/vuln/php_easter_egg/engine.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+            kill_switch += 1
+            try:
+                if threading.activeCount() is 1:
+                    break
+            except KeyboardInterrupt:
+                break
+        thread_write = int(open(thread_tmp_filename).read().rsplit()[0])
+        if thread_write is 1 and verbose_level is not 0:
+            info(messages(language, "not_found"))
+            data = json.dumps({'HOST': target, 'USERNAME': '', 'PASSWORD': '', 'PORT': '', 'TYPE': 'php_easter_eggs_vuln_scan',
```
Comment (securestep9) [inline lib/vuln/php_easter_egg/engine.py:192]: @aman566 please use the same name you started with 'php_easter_egg_vuln_scan' (not 'eggs')

File: lib/vuln/php_easter_egg/engine.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+            port = int(port)
+            t = threading.Thread(target=__php_easter_eggs_vuln,
+                                 args=(target, int(port), timeout_sec, log_in_file, language, time_sleep,
+                                       thread_tmp_filename, extra_requirements, socks_proxy, scan_id, scan_cmd))
+            threads.append(t)
+            t.start()
+            trying += 1
+            if verbose_level > 3:
+                info(
+                    messages(language, "trying_message").format(trying, total_req, num, total, target, port, 'php_easter_eggs_vuln_scan'))
```
Comment (securestep9) [inline lib/vuln/php_easter_egg/engine.py:165]: @aman566 please use the same name you started with 'php_easter_egg_vuln_scan' (not 'eggs')

File: lib/vuln/php_easter_egg/engine.py [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+        threads = []
+        total_req = len(ports)
+        thread_tmp_filename = '{}/tmp/thread_tmp_'.format(load_file_path()) + ''.join(
+            random.choice(string.ascii_letters + string.digits) for _ in range(20))
+        __log_into_file(thread_tmp_filename, 'w', '1', language)
+        trying = 0
+        keyboard_interrupt_flag = False
+        for port in ports:
+            port = int(port)
+            t = threading.Thread(target=__php_easter_eggs_vuln,
```
Comment (securestep9) [inline lib/vuln/php_easter_egg/engine.py:157]: @aman566 please use the same name you started with 'php_easter_egg_vuln_scan' (not 'eggs')

File: lib/vuln/php_easter_egg/engine.py [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+    if target_type(target) != 'SINGLE_IPv4' or target_type(target) != 'DOMAIN' or target_type(target) != 'HTTP':
+        # requirements check
+        new_extra_requirements = extra_requirements_dict()
+        if methods_args is not None:
+            for extra_requirement in extra_requirements_dict():
+                if extra_requirement in methods_args:
+                    new_extra_requirements[extra_requirement] = methods_args[extra_requirement]
+        extra_requirements = new_extra_requirements
+        if ports is None:
+            ports = extra_requirements["php_easter_eggs_vuln_ports"]
```
Comment (securestep9) [inline lib/vuln/php_easter_egg/engine.py:145]: @aman566 please use the same name you started with 'php_easter_egg_vuln_scan' (not 'eggs')

File: lib/vuln/php_easter_egg/engine.py [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+            else:
+                return False
+    except Exception as e:
+        print(e)
+        return False
+
+
+def __php_easter_eggs_vuln(target, port, timeout_sec, log_in_file, language, time_sleep,
+                     thread_tmp_filename, extra_requirements, socks_proxy, scan_id, scan_cmd):
+    if php_easter_eggs_vuln(target, port, timeout_sec, log_in_file, language, time_sleep,
```
Comment (securestep9) [inline lib/vuln/php_easter_egg/engine.py:120]: @aman566 please use the same name you started with 'php_easter_egg_vuln_scan' (not 'eggs')

File: lib/vuln/php_easter_egg/engine.py [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+                php_easter_egg = "PHP easter egg detected"
+                return True
+            else:
+                return False
+    except Exception as e:
+        print(e)
+        return False
+
+
+def __php_easter_eggs_vuln(target, port, timeout_sec, log_in_file, language, time_sleep,
```
Comment (securestep9) [inline lib/vuln/php_easter_egg/engine.py:118]: @aman566 please use the same name you started with 'php_easter_egg_vuln_scan' (not 'eggs')

File: lib/vuln/php_easter_egg/engine.py [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sys.stdout.flush()
+        s.settimeout(timeout_sec)
+        s.connect((targ, port))
+        return s
+    except Exception:
+        return None
+
+
+def php_easter_eggs_vuln(target, port, timeout_sec, log_in_file, language, time_sleep,
```
Comment (securestep9) [inline lib/vuln/php_easter_egg/engine.py:67]: @aman566 please use the same name you started with 'php_easter_egg_vuln_scan' (not 'eggs')

File: lib/vuln/php_easter_egg/engine.py [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+            if target_type(target) != "HTTP" and port == 80:
+                target = 'http://' + target
+            if(not target.endswith("/")):
+                req_php_credits_url = target + "/" + extra_requirements["php_easter_eggs"][0]
+                req_php_logo_url = target + "/" + extra_requirements["php_easter_eggs"][1]
+                req_zend_url = target + "/" + extra_requirements["php_easter_eggs"][2]
+                req_php_logo2_url = target + "/" + extra_requirements["php_easter_eggs"][3]
+            try:
+                user_agent = {'User-agent': random.choice(user_agent_list)}
+                req_php_credits = requests.get(req_php_credits_url, timeout=10, headers=user_agent)
```
Comment (securestep9) [inline lib/vuln/php_easter_egg/engine.py:90]: @aman566 please do not hardcode the timeout - it should be timeout_sec also verify=False is missing

File: lib/vuln/php_easter_egg/engine.py [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+                user_agent = {'User-agent': random.choice(user_agent_list)}
+                req_php_logo = requests.get(req_php_logo_url, timeout=10, headers=user_agent)
```
Comment (securestep9) [inline lib/vuln/php_easter_egg/engine.py:92]: @aman566 please do not hardcode the timeout - it should be timeout_sec also verify=False is missing

File: lib/vuln/php_easter_egg/engine.py [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+                user_agent = {'User-agent': random.choice(user_agent_list)}
+                req_zend = requests.get(req_zend_url, timeout=10, headers=user_agent)
```
Comment (securestep9) [inline lib/vuln/php_easter_egg/engine.py:94]: @aman566 please do not hardcode the timeout - it should be timeout_sec also verify=False is missing

File: lib/vuln/php_easter_egg/engine.py [hunk 12]
```diff
@@ -0,0 +1,201 @@
...
+                user_agent = {'User-agent': random.choice(user_agent_list)}
+                req_php_logo2 = requests.get(req_php_logo2_url, timeout=10, headers=user_agent)
```
Comment (securestep9) [inline lib/vuln/php_easter_egg/engine.py:96]: @aman566 please do not hardcode the timeout - it should be timeout_sec also verify=False is missing

**Review by securestep9 — CHANGES_REQUESTED** [review]

**Comment by Ali-Razmjoo** [comment]
this can't be supported anymore. modules are yaml #440

**Pull request closed without merging**